### PR TITLE
Simple Canonicalization

### PR DESCRIPTION
**Pull request by geneshuman**
Support for simple canonicalization.  This should be reviewed and merged after the as_fixes branch is in.  Or you can just look explicitly at the most recent commit.

### CODE REVIEW

File: libopenarc/arc.c [hunk 1]
```diff
@@ -1670,7 +1671,7 @@ arc_process_set(ARC_MESSAGE *msg, arc_kvsettype_t type, u_char *str,
 }
 
 /*
-**  ARC_GET_KEY -- acquire a public key used for verification
+**  ARC_GpET_KEY -- acquire a public key used for verification
```
Comment (mskucherawy) [inline libopenarc/arc.c]: "p"

File: openarc/openarc.c [hunk 2]
```diff
@@ -1768,16 +1768,18 @@ arcf_config_setlib(struct arcf_config *conf, char **err)
 	{
 		char *s = conf->conf_signhdrs_raw;
 		int cnt;
-		for (cnt=0; s[cnt]; s[cnt]==',' ? cnt++ : *s++);
+		for (cnt=0;
+		     s[cnt];
+		     s[cnt]==',' ? cnt++ : *s++);
```
Comment (mskucherawy) [inline openarc/openarc.c]: Should be immediately followed by "continue", rather than having that be implicit.

File: openarc/openarc.c [hunk 3]
```diff
@@ -1798,16 +1800,18 @@ arcf_config_setlib(struct arcf_config *conf, char **err)
 	{
 		char *s = conf->conf_oversignhdrs_raw;
 		int cnt;
-		for (cnt=0; s[cnt]; s[cnt]==',' ? cnt++ : *s++);
+		for (cnt=0;
+		     s[cnt];
+		     s[cnt]==',' ? cnt++ : *s++);
```
Comment (mskucherawy) [inline openarc/openarc.c]: "continue"

File: libopenarc/arc-canon.c [hunk 4]
```diff
@@ -2096,3 +2096,42 @@ arc_canon_add_to_seal(ARC_MESSAGE *msg)
...
+ARC_STAT
+arc_parse_canon_t(unsigned char *tag, arc_canon_t *hdr_canon, arc_canon_t *body_canon)
+{
+	char *token;
+	int code;
+
+	token = strtok(tag, "/");
+	code = arc_name_to_code(canonicalizations, token);
+
+	if(code == -1)
```
Comment (mskucherawy) [inline libopenarc/arc-canon.c]: Space after "if".

File: libopenarc/arc-canon.c [hunk 5]
```diff
@@ -2096,3 +2096,42 @@ arc_canon_add_to_seal(ARC_MESSAGE *msg)
...
+		return ARC_STAT_INVALID;
+
+	*hdr_canon = (arc_canon_t)code;
```
Comment (mskucherawy) [inline libopenarc/arc-canon.c]: Space after a cast.

File: libopenarc/arc-canon.c [hunk 6]
```diff
@@ -2096,3 +2096,42 @@ arc_canon_add_to_seal(ARC_MESSAGE *msg)
...
+
+	token = strtok(NULL, "/");
+	code = arc_name_to_code(canonicalizations, token);
+
+	if(code == -1)
```
Comment (mskucherawy) [inline libopenarc/arc-canon.c]: Space after "if".

File: libopenarc/arc-canon.c [hunk 7]
```diff
@@ -2096,3 +2096,42 @@ arc_canon_add_to_seal(ARC_MESSAGE *msg)
...
+		return ARC_STAT_INVALID;
+
+	*body_canon = (arc_canon_t)code;
```
Comment (mskucherawy) [inline libopenarc/arc-canon.c]: Space after a cast.

File: libopenarc/arc.c [hunk 8]
```diff
@@ -2525,14 +2526,14 @@ arc_eoh(ARC_MESSAGE *msg)
...
-	**  Request specific canonicalizations we want to run.
+	** Request specific canonicalizations we want to run.
 	*/
 
-	/* headers, validation */
 	h = NULL;
 	htag = NULL;
 	if (nsets > 0)
 	{
+  	/* headers, validation */
```
Comment (mskucherawy) [inline libopenarc/arc.c]: Fix indent.

**Comment by geneshuman** [comment]
All of the requested format changes have been made.  Note, this diff should be reviewed & merged after the AS_fixes patch

**Comment by mskucherawy** [comment]
Obsoleted by other PRs.

**Pull request closed without merging**